dashboard/config/routes.rb

```diff
@@ -754,4 +754,13 @@
   get '/form/:misc_form_path', to: 'foorm/misc_survey#new'
 
   post '/i18n/track_string_usage', action: :track_string_usage, controller: :i18n
+
+  namespace :foorm do
+    resources :forms, only: [:create] do
+      member do
+        put :edit_questions
+        put :publish
```

Comment:
This cleaned up nicely!

I think it could be fine to keep `publish` separate here, if you want to have a clear division between editing questions and updating.

`edit_questions` seems like it could be turned into `update`, unless there is a particular reason not to. also, `publish` _could_ be rolled into edit, but I can see why you might want to keep this sensitive operation separate.

do you think you will want an `edit` route which loads the edit page that would be used to make PUT requests to `edit_questions` or `update`?

Reply:
My main reason for `edit_questions` instead of `update` was just to make it clear we only allow editing questions, not name/version or published. Not super opinionated on the terminology though. I also wanted the edit and publish steps to be separate just for clarity's sake. For now I don't think we will want the `edit` route, since the editor loads up a form without loading a new page, but I might need to think more on that.

Reply:
ok, that makes sense to me, thanks! I am looking at the rest of the PR now.

Reply:
one more request here - I would suggest calling this `update_questions` then, since it is more like `update` than `edit`.